src/UI/ModalWindow.jsx

```diff
@@ -0,0 +1,200 @@
+  return fn && {}.toString.call(fn) === '[object Function]';
+ }
+
+const ModalWindow = ({exitHandler, modaldata}) => {
+  const {exit, error, type, icon, title, description, inputId, firstButton, secondButton, submit} = modaldata;
+
+  const [username, handleUsername] = useState();
+  const [password, handlePassword] = useState();
+  const [confirm, handleConfirm] = useState();
+
+  return (
+    <ReactModal appElement={document.getElementById('root')} isOpen={true} style={ModalStyles}>
+      {exit && <Exit type="button" onClick={(e) => exitHandler()}>✖</Exit>}
+      {icon && <ModalIcon src={icon} alt='icon'></ModalIcon>}
+      {title && <ModalTitle type={type}>{title}</ModalTitle>}
+      {description && <ModalDescription type={type}>{description}</ModalDescription>}
+      { ((type === "login" || type === "register" || type === 'input') &&
+        <form onSubmit={(e) => submit(e, {username, password, confirm})}>
+          { type === 'input' &&
+            <ModalInput type="text" id={inputId}></ModalInput>
+          }
+          { (type === "login" || type === "register") &&
+            <InputLabel>
+              <Label htmlFor="username">Username</Label>
+              <ModalInput id="username" name="username" required onChange={(e) => handleUsername(e.target.value)} type="text"></ModalInput>
+            </InputLabel>
+          }
+          { (type === "login" || type === "register") &&
+            <InputLabel>
+              <Label htmlFor="password">Password</Label>
+              <ModalInput id="password" name="password" required onChange={(e) => handlePassword(e.target.value)} type="password"></ModalInput>
+            </InputLabel>
+          }
+          { type === "register" &&
+            <InputLabel>
+              <Label htmlFor="password_confirm">Confirm password</Label>
+              <ModalInput id="password_confirm" name="password" required onChange={(e) => handleConfirm(e.target.value)} type="password"></ModalInput>
+            </InputLabel>
+          }
+          { error &&
+            <Error>{error}</Error>
+          }
+          { (type === "login" || type === "register") &&
+            <Social>
+              or login with social account
+              <a href="https://github.com/login/oauth/authorize?client_id=70e40fe40ada41351efa"><SocialIcon src={github} alt="github" style={{filter: "brightness(0)"}}/></a>
+            </Social>
+          }
+          <Buttons>
+            {
+              <ModalButton type="submit">{firstButton.text}</ModalButton>
+            }
+            {
+            secondButton.text &&
+              <ModalButton type="button" onClick={(e) => secondButton.handler()}>{secondButton.text}</ModalButton>
+            }
+          </Buttons>
+        </form> )
+      ||
+      <Buttons>
+        { firstButton.text &&
+          <ModalButton type="button" onClick={(e) => isFunction(firstButton.handler) ? firstButton.handler() : false}>{firstButton.text}</ModalButton>
```

Comment:
Why do you need to return a `false` from click handler? Otherwise use `isFunction(firstButton.handler) && firstButton.handler()`